core/llm/autodetect.ts

```diff
@@ -76,7 +76,9 @@ function modelSupportsImages(
   provider: ModelProvider,
   model: string,
   title: string | undefined,
+  capability: ModelCapability | undefined
```

Comment:
Since we are only checking for the support images property, I believe we could just do something like this:

```suggestion
  imageUploadCapability: ModelCapability['uploadImage']
```

And over here, pass in `this.capability.uploadImage`

https://github.com/continuedev/continue/pull/1810/files#diff-e3b0303fc757a51f78ed6211349a5f7f34c28167a821b77a45faa4abd6468358R57